server/kit/kitserver.go

```diff
@@ -0,0 +1,178 @@
+package kit
+
+import (
+	"context"
+	"crypto/tls"
+	"fmt"
+	"net"
+	"net/http"
+	"os"
+	"time"
+
+	ocontext "golang.org/x/net/context"
+	"google.golang.org/grpc"
+
+	gserver "github.com/NYTimes/gizmo/server"
+	"github.com/go-kit/kit/log"
+	httptransport "github.com/go-kit/kit/transport/http"
+	grpc_middleware "github.com/grpc-ecosystem/go-grpc-middleware"
+)
+
+type server struct {
+	logger log.Logger
+
+	mux Router
+
+	cfg gserver.Config
+
+	svc Service
+
+	// exit chan for graceful shutdown
+	exit chan chan error
+}
+
+// newServer will init the mux and register all endpoints.
+func newServer(cfg gserver.Config, opts ...RouterOption) *server {
+	if len(opts) == 0 {
+		// select the default router
+		opts = append(opts, RouterSelect(""))
+	}
+	var r Router
+	for _, opt := range opts {
+		r = opt(r)
+	}
+	return &server{
+		cfg:    cfg,
+		mux:    r,
+		exit:   make(chan chan error),
+		logger: log.NewJSONLogger(log.NewSyncWriter(os.Stdout)),
+	}
+}
+
+func (s *server) ServeHTTP(w http.ResponseWriter, r *http.Request) {
+	// hand the request off to the router
+	s.mux.ServeHTTP(w, r)
+}
+
+// Register will accept and register server, JSONService or MixedService implementations.
+func (s *server) Register(svc Service) error {
+	s.svc = svc
+	opts := []httptransport.ServerOption{
+		// inject the server logger into every request context
+		httptransport.ServerBefore(func(ctx context.Context, _ *http.Request) context.Context {
+			return context.WithValue(ctx, logKey, s.logger)
+		}),
+	}
+	opts = append(opts, defaultHTTPOpts...)
+	opts = append(opts, svc.HTTPOptions()...)
+
+	// register all endpoints with our wrappers & default decoders/encoders
+	for path, epMethods := range svc.HTTPEndpoints() {
+		for method, ep := range epMethods {
+			// just pass the http.Request in if no decoder provided
+			if ep.Decoder == nil {
+				ep.Decoder = func(_ context.Context, r *http.Request) (interface{}, error) {
+					return r, nil
+				}
+			}
+			// default to the httptransport helper
+			if ep.Encoder == nil {
+				ep.Encoder = httptransport.EncodeJSONResponse
+			}
+			s.mux.Handle(method, path, svc.HTTPMiddleware(
+				httptransport.NewServer(
+					svc.Middleware(ep.Endpoint),
+					ep.Decoder,
+					ep.Encoder,
+					append(opts, ep.Options...)...)))
+		}
+	}
+
+	return nil
+}
+
+func (s *server) Start() error {
+	// TODO(jprobinson) add basic default health check
+
+	// HTTP SERVER!
+	addr := fmt.Sprintf(":%d", s.cfg.HTTPPort)
```

Comment:
This is meant to be in a container, DNS should be terminated by the time it gets to this point.